metadata/metadata_service/proxy/neo4j_proxy.py

```diff
@@ -520,14 +539,15 @@ def _safe_get(self, dct, *keys):
     @timer_with_counter
     def _execute_cypher_query(self, *,
                               statement: str,
-                              param_dict: Dict[str, Any]) -> BoltStatementResult:
+                              param_dict: Dict[str, Any]) -> List[Record]:
         if LOGGER.isEnabledFor(logging.DEBUG):
             LOGGER.debug('Executing Cypher query: {statement} with params {params}: '.format(statement=statement,
                                                                                              params=param_dict))
         start = time.time()
         try:
-            with self._driver.session() as session:
-                return session.run(statement, **param_dict)
+            with self._driver.session(database=self._database_name) as session:
+                result = session.run(query=statement, **param_dict)
+                return [record for record in result]
```

Comment:
why. IIUC result is not actually a list but it does already implement the list interface methods, so i don't see a reason for this.

Reply:
result needs to be consumed within the session or it ends up being []

Reply:
found something instructing to do exactly this: https://community.neo4j.com/t5/drivers-stacks/python-neo4j-driver-get-result-from-session-run/td-p/13567
Thanks!